hack to get semicolon through to API

jduss4 opened the pull request:
Regardless of whether the api_bridge is receiving a literal semicolon in the options list, or whether it is receiving an encoded (`%3B`) version, make sure that it is sent to the API as `%3B` in order to be recognized as `;`

If we decide to post data to the API instead of via the URL, this could be revisited.

Changes:
- bumps a version of the rails gem
- writes a test for the semicolon step
- removes unused method

Needs to be done (creating issues for these):
- revisit URI library in favor of CGI
- redo syntax throughout gem to match internal CDRH styleguide choices

jduss4 committed:
hack to get semicolon through to API

techgique commented:
Looks great.

I'd really like to try to get the change for the obsolete method in before we finalize for initial launch 🤞 
Otherwise, I worry we'll be too paranoid to make the change once we have something live and it'll be a big ordeal with version changes etc.

Seems like you found this also, but here's a link for use with creating the issue as mentioned above: https://ruby-doc.org/stdlib-2.4.2/libdoc/uri/rdoc/URI/Escape.html#method-i-escape-label-Description

I'll merge once those issues are created :)

techgique committed:
Replace obsolete URI.encode; Update tests

Supported encoding methods encode far more characters
Split URL passed into individual components
Encode only param values
Skip encoding param names because the [] encoding breaks API requests

Must pass a full URL in tests to use method which splits components
Update tests to reflect different encoding output
- Spaces encode as '+', commas as '%2C'
Remove test for pre-encoded semi-colons
Update assert_equal argument order for modified tests

techgique commented:
https://github.com/CDRH/api_bridge/tree/replace-obsolete-encoding addresses #2 

Seems to make semicolons work without the pre-encoding of semicolons in Orchid PR https://github.com/CDRH/orchid/pull/57, so may need to remove that change. Retaining the minimal code cleanup would be nice though :)

techgique reviewed (changes requested):
Give my branch a look and see if it behaves for you as well 🤞

jduss4 committed:
simplifies url encoding and adds a few tests
